sns_ik_examples/src/ik_tests.cpp

```diff
@@ -60,20 +60,39 @@ double getDeltaWithLimits(double value, double desired_delta,
   }
 }
 
-bool in_vel_bounds(KDL::JntArray vals, KDL::JntArray vels)
+bool in_vel_bounds(const KDL::JntArray& vals, const KDL::JntArray& vels)
 {
   for(size_t i; i < vels.data.size(); i++){
-      if(vals(i) < -vels(i) || vals(i) > vels(i)){
+      if(vals(i)-1e-6 < -vels(i) || vals(i) > vels(i)+1e-6){
```

Comment:
The first condition should be (vals(i) < -vels(i) - 1e-6).  Also the name vels is non-descriptive and really close to vals.